lib/data/model/data_state.dart

```diff
@@ -0,0 +1,21 @@
+abstract class DataState<T> {
+  final T? data;
+  final String? errorMsg;
+
+  const DataState({
+    this.data,
+    this.errorMsg,
+  });
+}
+
+class DataSuccess<T> extends DataState<T> {
+  const DataSuccess(T data) : super(data: data);
+}
+
+class DataFailed<T> extends DataState<T> {
+  const DataFailed(String error) : super(errorMsg: error);
+}
+
+class DataNotSet<T> extends DataState {
+  const DataNotSet();
+}
```

Comment:
This is like a DataSuccess with empty data, right?

Reply:
yes Fran!